Panic when completing adaptive chores with nil PerformedAt in history

## Summary
When completing a chore with `FrequencyType = "adaptive"`, the application panics if there are history entries with nil `PerformedAt` values.

## Error Details
```
panic: runtime error: invalid memory address or nil pointer dereference
goroutine 19746 [running]:
donetick.com/core/internal/chore.scheduleAdaptiveNextDueDate(0xc00012c500, {0x14410450, 0xee0385af4, 0x0}, {0xc0008baf00, 0x3, 0x7fd6cef61438?})
    /home/runner/work/donetick/donetick/internal/chore/scheduler.go:171 +0x548
```

## Root Cause
In `scheduleAdaptiveNextDueDate` function, the code attempts to access `history[i].PerformedAt.UTC()` without checking if `PerformedAt` is nil:

```go
delay := history[i].PerformedAt.UTC().Sub(history[i+1].PerformedAt.UTC()).Seconds()
```

## Steps to Reproduce
1. Have a chore with `frequency_type = "adaptive"`
2. Have chore history entries where `performed_at` is NULL
3. Try to complete the chore via API or UI
4. Application crashes with nil pointer dereference

## Expected Behavior
The application should handle nil `PerformedAt` values gracefully, either by:
- Skipping entries with nil timestamps
- Using a default value
- Returning an appropriate error message

## Actual Behavior
Application panics and returns 503 Service Unavailable

--- a/internal/chore/scheduler.go
+++ b/internal/chore/scheduler.go
@@ -168,10 +168,24 @@ func scheduleAdaptiveNextDueDate(chore *chModel.Chore, completedDate time.Time, 
 	decayFactor := 0.5 // Adjust this value to control the decay rate
 
 	for i := 0; i < len(history)-1; i++ {
+		// Skip entries with nil PerformedAt
+		if history[i].PerformedAt == nil || history[i+1].PerformedAt == nil {
+			continue
+		}
 		delay := history[i].PerformedAt.UTC().Sub(history[i+1].PerformedAt.UTC()).Seconds()
 		weight := math.Pow(decayFactor, float64(i))
 		totalDelay += delay * weight
 		totalWeight += weight
+	}
+
+	// If no valid history entries, fall back to default behavior
+	if totalWeight == 0 {
+		if chore.NextDueDate != nil {
+			diff := completedDate.UTC().Sub(chore.NextDueDate.UTC())
+			nextDueDate := completedDate.UTC().Add(diff)
+			return &nextDueDate, nil
+		}
+		return nil, nil
 	}
 
 	averageDelay := totalDelay / totalWeight

--- a/internal/chore/scheduler_adaptive_test.go
+++ b/internal/chore/scheduler_adaptive_test.go
@@ -1,0 +1,130 @@
+package chore
+
+import (
+	"testing"
+	"time"
+
+	chModel "donetick.com/core/internal/chore/model"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestScheduleAdaptiveNextDueDate_WithNilPerformedAt(t *testing.T) {
+	now := time.Now().UTC()
+	
+	tests := []struct {
+		name          string
+		chore         *chModel.Chore
+		completedDate time.Time
+		history       []*chModel.ChoreHistory
+		wantNil       bool
+		wantErr       bool
+	}{
+		{
+			name: "All history entries have nil PerformedAt",
+			chore: &chModel.Chore{
+				FrequencyType: chModel.FrequencyTypeAdaptive,
+				NextDueDate:   timePtr(now.Add(24 * time.Hour)),
+			},
+			completedDate: now,
+			history: []*chModel.ChoreHistory{
+				{PerformedAt: nil},
+				{PerformedAt: nil},
+				{PerformedAt: nil},
+			},
+			wantNil: false,
+			wantErr: false,
+		},
+		{
+			name: "Mixed nil and valid PerformedAt entries",
+			chore: &chModel.Chore{
+				FrequencyType: chModel.FrequencyTypeAdaptive,
+			},
+			completedDate: now,
+			history: []*chModel.ChoreHistory{
+				{PerformedAt: timePtr(now.Add(-24 * time.Hour))},
+				{PerformedAt: timePtr(now.Add(-48 * time.Hour))},
+				{PerformedAt: nil},
+				{PerformedAt: timePtr(now.Add(-96 * time.Hour))},
+			},
+			wantNil: false, // Should calculate from valid consecutive entries
+			wantErr: false,
+		},
+		{
+			name: "Single history entry with nil PerformedAt",
+			chore: &chModel.Chore{
+				FrequencyType: chModel.FrequencyTypeAdaptive,
+				NextDueDate:   timePtr(now.Add(24 * time.Hour)),
+			},
+			completedDate: now,
+			history: []*chModel.ChoreHistory{
+				{PerformedAt: nil},
+			},
+			wantNil: false,
+			wantErr: false,
+		},
+		{
+			name: "Empty history with NextDueDate",
+			chore: &chModel.Chore{
+				FrequencyType: chModel.FrequencyTypeAdaptive,
+				NextDueDate:   timePtr(now.Add(24 * time.Hour)),
+			},
+			completedDate: now,
+			history:       []*chModel.ChoreHistory{},
+			wantNil:       false,
+			wantErr:       false,
+		},
+		{
+			name: "Empty history without NextDueDate",
+			chore: &chModel.Chore{
+				FrequencyType: chModel.FrequencyTypeAdaptive,
+				NextDueDate:   nil,
+			},
+			completedDate: now,
+			history:       []*chModel.ChoreHistory{},
+			wantNil:       true,
+			wantErr:       false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := scheduleAdaptiveNextDueDate(tt.chore, tt.completedDate, tt.history)
+			
+			if tt.wantErr {
+				assert.Error(t, err)
+				return
+			}
+			
+			assert.NoError(t, err)
+			
+			if tt.wantNil {
+				assert.Nil(t, got)
+			} else {
+				assert.NotNil(t, got)
+			}
+		})
+	}
+}
+
+// Test that the fix doesn't break normal adaptive scheduling
+func TestScheduleAdaptiveNextDueDate_NormalCase(t *testing.T) {
+	now := time.Now().UTC()
+	
+	chore := &chModel.Chore{
+		FrequencyType: chModel.FrequencyTypeAdaptive,
+	}
+	
+	history := []*chModel.ChoreHistory{
+		{PerformedAt: timePtr(now.Add(-24 * time.Hour))},
+		{PerformedAt: timePtr(now.Add(-48 * time.Hour))},
+		{PerformedAt: timePtr(now.Add(-72 * time.Hour))},
+	}
+	
+	got, err := scheduleAdaptiveNextDueDate(chore, now, history)
+	
+	assert.NoError(t, err)
+	assert.NotNil(t, got)
+	// The next due date should be approximately 24 hours from now
+	// (based on the pattern in history)
+	assert.InDelta(t, now.Add(24*time.Hour).Unix(), got.Unix(), 3600) // within 1 hour
+}

